crates/sui-core/src/authority_server.rs

```diff
@@ -415,6 +416,12 @@ impl ValidatorService {
                 .await
                 .map_err(|e| tonic::Status::internal(e.to_string()))?
         {
+            // Note that num_inflight_transactions() only include user submitted transactions, and only user txns can be dropped here.
+            // This backpressure should not affect system transactions, e.g. for checkpointing.
+            if consensus_adapter.num_inflight_transactions() > MAX_PENDING_CONSENSUS_TRANSACTIONS {
+                return Err(tonic::Status::resource_exhausted("Reached {MAX_PENDING_CONSENSUS_TRANSACTIONS} concurrent consensus transactions",
```

Comment:
nit: we probably want a counter here so that we know how many txns we are rejecting

Reply:
If there is no general gRPC status metrics, will add that in a follow up.